desafio-07/MatMercer/go/tac.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+func main() {
+	tac()
+}
+
+// in MB size, currently the program uses a max of 2*bufSize iff file size > bufSize
+const maxBufSize = int64(250 << (10 * 2))
+
+var stdout *bufio.Writer
+
+func check(e error) {
+	if e != nil {
+		errAndExit(fmt.Sprintf("tac error: %v", e))
+	}
+}
+
+func errAndExit(msg string) {
+	_, _ = os.Stderr.WriteString(msg + "\n")
+	os.Exit(1)
+}
+
+func tac() {
+	args := os.Args[1:]
+
+	if len(args) != 1 {
+		errAndExit("usage: tac [file]")
+	}
+
+	fileName := args[0]
+
+	f, err := os.Open(fileName)
+	check(err)
```

Comment:
Nooooope :)

Error checking é sempre inline. Lembre-se da regra: Dentro do main, um error failure sai do programa com um `log.Fatal` (se for um erro fatal). Dentro de funções, sempre retorne o erro pro caller, nunca use `log.Fatal` ou saia do programa (a não ser em alguns casos super específicos).

Aqui seria (e em vários outros lugares no código), seria:

```
if err != nil {
    log.Fatal(err)
}
```